### PR TITLE
Add library logic

### PR DESCRIPTION
**Pull request by jinhanfromNUS**
Books are now stored in the txt file in the file path **"./library.txt"**

Q) How to create a library object with all the available books loaded to the file?
LibraryLogic lb = new LibraryLogic();
Library library = new Library(lb.getAvailableBooksFromFile());

Q) How to save the available books into the txt file?
LibraryLogic lb = new LibraryLogic();
lb.saveBooksToFile(ArrayList<Book>);

Note:
1. I am doing this assuming the txt file name and path is **"./library.txt"**. We can change it if needed.
2. We actually can implement this directly within the Library Class. We can change if needed.
3. LibraryLogic has two constructors:
* LibraryLogic() --> create a LibraryLogic object with default file path
* LibrayLogic(String filePath) --> create a LibraryLogic object with parsed file path (we can delete this if we decided not to allow the Librarian to change the txt file name and storage)
4. I **haven't** merge Suraj's code.

### CODE REVIEW

File: src/main/java/seedu/address/model/library/LibraryLogic.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    /**
+     * Constructs a LibraryLogic object with default file path
+     */
+    public LibraryLogic() {
+        this.filePath = "./library.txt";
+        this.availableBooks = new ArrayList<>();
+    }
+
+    /**
+     * Validates if the start date is before the end date.
```
Comment (ashleyclx) [inline src/main/java/seedu/address/model/library/LibraryLogic.java:40]: Header comment is incorrect, should be checking if valid book title not date

File: src/main/java/seedu/address/model/library/LibraryLogic.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+     *
+     * @param bookTitle the book title of a book
+     * @return true if the start date is before the end date, false otherwise
```
Comment (ashleyclx) [inline src/main/java/seedu/address/model/library/LibraryLogic.java:43]: Same thing for return value comment incorrect

File: src/main/java/seedu/address/model/library/LibraryLogic.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            while ((line = reader.readLine()) != null) {
+                if (checkValidBook(line.trim())) {
+                    // todo throw an exception here
+                    System.out.println("Error found: Bad input");
+                }
+                Book currentBook = new Book(line.trim());
+                availableBooks.add(currentBook);
+            }
+        } catch (IOException e) {
+            // todo throw an exception here
```
Comment (ashleyclx) [inline src/main/java/seedu/address/model/library/LibraryLogic.java:88]: Potential area to use logger for errors

File: src/main/java/seedu/address/model/library/LibraryLogic.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                System.out.println("An error occurred while creating the file.");
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Loads books from the file and store it inside the availableBooks.
+     */
+    public void loadBooksFromFile() {
```
Comment (ashleyclx) [inline src/main/java/seedu/address/model/library/LibraryLogic.java:74]: Potential place to use logger when loading books

**Review by ashleyclx — APPROVED** [review]
LGTM! A few nits to pick!

File: src/main/java/seedu/address/model/library/LibraryLogic.java (hunk 1 above)
Comment (ashleyclx) [reply src/main/java/seedu/address/model/library/LibraryLogic.java:40]: I'll fix this

File: src/main/java/seedu/address/model/library/LibraryLogic.java (hunk 2 above)
Comment (ashleyclx) [reply src/main/java/seedu/address/model/library/LibraryLogic.java:43]: I'll fix this

File: src/main/java/seedu/address/model/library/LibraryLogic.java (hunk 4 above)
Comment (ashleyclx) [reply src/main/java/seedu/address/model/library/LibraryLogic.java:74]: Nevermind should be in during init()

File: src/main/java/seedu/address/model/library/LibraryLogic.java (hunk 3 above)
Comment (ashleyclx) [reply src/main/java/seedu/address/model/library/LibraryLogic.java:88]: Nevermind should be during init()

File: src/main/java/seedu/address/model/library/LibraryLogic.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        try (PrintWriter writer = new PrintWriter(new FileWriter(filePath))) {
+            for (Book availableBook : library) {
+                writer.println(availableBook);
+            }
+        } catch (IOException e) {
+            // todo throw an exception here
+            System.err.println("Error saving books to file: " + e.getMessage());
+        }
+    }
+}
```
Comment (Ty-stan0417) [inline src/main/java/seedu/address/model/library/LibraryLogic.java:120]: LGTM

File: src/main/java/seedu/address/model/library/LibraryLogic.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                    // todo return a String instead
+                    System.out.println("File creation failed.");
+                }
+            } catch (IOException e) {
+                // todo create an exception here
+                System.out.println("An error occurred while creating the file.");
+                e.printStackTrace();
+            }
+        }
+    }
```
Comment (Ty-stan0417) [inline src/main/java/seedu/address/model/library/LibraryLogic.java:69]: Good as a separate method to avoid long method

**Review by Ty-stan0417 — COMMENTED** [review]
LGTM

**Pull request merged**